spark3-extensions/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveProcedures.scala

```diff
@@ -79,19 +79,7 @@ case class ResolveProcedures(spark: SparkSession) extends Rule[LogicalPlan] with
 
     nameToArgMap.foreach { case (name, arg) =>
       val position = nameToPositionMap(name)
-      val param = params(position)
-      val paramType = param.dataType
-      val argType = arg.expr.dataType
```

Comment:
We cannot call `dataType` as it may not be resolved yet (e.g. calls to functions).